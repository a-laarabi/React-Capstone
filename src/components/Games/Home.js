/* eslint-disable no-nested-ternary */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setGames } from '../../redux/actions';
import GamesItem from './GameItem';

const Games = () => {
  const games = useSelector((state) => state.games);
  const dispatch = useDispatch();
  dispatch(setGames());

  const filteredGames = games.filter((game) => game.title.toLowerCase());

  return (
    <>
      <div className="content">
        {
            filteredGames.length > 0 ? (
              filteredGames.map((game) => (
                <GamesItem
                  key={game.id}
                  id={game.id}
                  title={game.title}
                  released={game.released}
                  image={game.image}
                  gener={game.gener}
                />
              ))
            ) : (
              <div>
                <h1>Loading..</h1>
              </div>
            )
          }
      </div>
    </>
  );
};

export default Games;
