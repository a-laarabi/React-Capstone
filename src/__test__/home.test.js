import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Home from '../components/Games/Home';

describe('Test for Missions components', () => {
  it('Check if the Missions section are available', () => {
    const website = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(website).toMatchSnapshot();
  });
});
