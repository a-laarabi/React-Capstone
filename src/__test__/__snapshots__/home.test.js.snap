// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Missions components Check if the Missions section are available 1`] = `
<div
  className="content"
>
  <div>
    <h1>
      Loading..
    </h1>
  </div>
</div>
`;
